### PR TITLE
Added documentation for variable.py

### PR DESCRIPTION
**Pull request by Josh-Bodner**
Added documentation to variable.py

### CODE REVIEW

File: autodiff/variable.py [hunk 1]
```diff
@@ -1,24 +1,63 @@
...
 
 class Variable:
     """ 
-    #TODO: Write some documentation about that.
-    Variable class carry the information flow within the computational graph. 
-    Data attribute, Gradient attribute.
-    Data represents the evaluation point and gradient is the gradient held 
-    by the variable. By default = 1. 
+    Represents a variable, and carries the information flow
+    within the computational graph. Under the current
```
Comment (tguens) [inline autodiff/variable.py]: A Variable is an object, not a scalar. It has two attributes val, grad that are numpy arrays (made of np.float64 type).
Sor far, it is supopsed to be instantiated with  a numeric type (e.g float, int) but it handles np.array directly, or even a 1-dim tuple or 1-d list. 
e.g : x = Variable(2.) 
Can handle x = Variable([2]) or x = Variable((2)) or x = Variable(np.array([2])).

File: autodiff/variable.py [hunk 2]
```diff
@@ -1,24 +1,63 @@
...
     
     def __repr__(self):
+        """ When variables are printed, gives both val and grad"""
         return "Value: {}\nGradient: {}".format(self.val, self.grad)
     
     def __add__(self, other):
+        """Implements addition (self + other) between Variables and
+            other objects, which are either Variables or numeric values. 
+    
+        INPUTS
```
Comment (tguens) [inline autodiff/variable.py:45]: Great

File: autodiff/variable.py [hunk 3]
```diff
@@ -1,24 +1,63 @@
...
         self.val = get_right_shape(val)
         #grad = np.ones((len(self.val), ))
         self.grad = get_right_shape(grad)
     
     def __repr__(self):
+        """ When variables are printed, gives both val and grad"""
         return "Value: {}\nGradient: {}".format(self.val, self.grad)
     
     def __add__(self, other):
+        """Implements addition (self + other) between Variables and
```
Comment (tguens) [inline autodiff/variable.py]: Can you do the same without writing "self+other" explictly. 
"Implements addition between.." or "overload..." are more than enough.

Comment (tguens) [reply autodiff/variable.py]: And same thing for __mul_ so on, it's better not to say "self, other"

File: autodiff/variable.py [hunk 4]
```diff
@@ -30,8 +69,28 @@ def __add__(self, other):
             return Variable(val=out_val, grad=out_grad)
 
     def __mul__(self, other):
-        #Multi-dim: should be np.dot
-        #male_sure_shape(self,other)
+        """Implements multiplication (self * other) between Variables and
```
Comment (tguens) [inline autodiff/variable.py]: Between variables.

**Review by Gavindeed — APPROVED** [review]
Approved

**Pull request merged**